sdk-extensions/autoconfigure/src/main/java/io/opentelemetry/sdk/autoconfigure/AutoConfiguredOpenTelemetrySdk.java

```diff
@@ -40,11 +41,15 @@ public static AutoConfiguredOpenTelemetrySdkBuilder builder() {
   }
 
   static AutoConfiguredOpenTelemetrySdk create(
-      OpenTelemetrySdk sdk, Resource resource, ConfigProperties config) {
+      @Nullable OpenTelemetrySdk sdk, Resource resource, ConfigProperties config) {
     return new AutoValue_AutoConfiguredOpenTelemetrySdk(sdk, resource, config);
   }
 
-  /** Returns the {@link OpenTelemetrySdk} that was auto-configured. */
+  /**
+   * Returns the {@link OpenTelemetrySdk} that was auto-configured, or {@code null} if the SDK has
+   * been disabled.
+   */
+  @Nullable
```

Comment:
What do you think about putting the flag in the API only for the implocit autoconfigure instead of autoconfigure actually?

https://github.com/open-telemetry/opentelemetry-java/blob/main/api/all/src/main/java/io/opentelemetry/api/GlobalOpenTelemetry.java#L214

The nullable here seems to reduce ux of autoconfigure significantly

Reply:
Also I believe the javaagent currently is designed with the assumption that it is either completely disabled, or has an SDK. Changing that by making this nullable will be fairly tricky actually. Probably (not certainly) many apps using autoconfigure would have similar downstream effects if autoconfiguration was requested but disabled.

I think it's probably best to leave this flag guarding of autoconfiguration itself to callers, though adding the flag for the implicit magic seems good. Alternatively, we would want to return an SDK without any exporters, etc, though we know that does still do more work than may be expected.

Reply:
Converting the null to no-op API isn't that bad but it's still a huge difference in UX. Before no flags really affected how to use autoconfigure, now this one requires the caller to do more work even if they're not interested in the feature. That work could have been defining a flag instead without a lot more effort and not affecting the default experience. This is an experimental flag that may not make it through spec anyways (perhaps other languages have even bigger problems than we have here). So it doesn't seem appropriate for it to affect the UX like this.

Reply:
It does feel strange to change the UX in such a significant way to accommodate an experimental property. But it does seem like the value of the feature is eroded quite a bit if it doesn't work for users of `AutoConfiguredOpenTelemetrySdk`. I'll see if I can think of any other options.

Reply:
I tried something different in a new commit. Instead of `getOpenTelemetrySdk()` returning null, I've added a new method called `AutoConfiguredOpenTelemetrySdk#isSdkEnabled()`. The ergonomics are unchanged for someone who is not using the experiemental property, but for someone who is, `GlobalOpenTelemetry` is still set to noop and they have a way to tell if the property was set to disable the SDK.

When `isSdkEnabled() == false`, the response of `getOpenTelemetrySdk()` should be ignored, but returns `OpenTelemetrySdk.builder().build()` for lack of a better option. 

Still not a perfect solution, but maybe good enough while balancing the various arguments.

Reply:
I'd say let's leave the isConfigured method out for now as it's new API for an experimental flag. We could add it in the future of the flag stabilizes, but I think the main problem of exporting data to a non-existent collector automatic is resolved by returning the empty SDK

Reply:
I've removed the method in the latest commit 👍